### PR TITLE
Animal example for NGS blog

### PR DESCRIPTION
**Pull request by brooksmtownsend**
Simple example with httpclient, httpserver, and public APIs to randomly display a cat or a dog (or something else if it fails)

This example also includes a `core` and `extender` config files for use with the blog

### CODE REVIEW

File: ngs/dogs-and-cats/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# DogsAndCats Actor
```
Comment (autodidaddict) [inline ngs/dogs-and-cats/README.md:1]: nit: this is from the template, can probably make it `Dogs and Cats`

File: ngs/dogs-and-cats/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+This actor implements a simple piece of functionality: source and display a random image of either a cat or a dog.
+
+This actor uses the following capabilities:
+1. `wasmcloud:httpserver` to receive HTTP requests
+1. `wasmcloud:httpclient` to make HTTP requests to public APIs for animal image URLs, and then query that URL for the picture
+1. `wasmcloud:builtin:numbergen` to generate a random number to decide if a cat or dog shoud be fetched
+
+The following commands can be run to start this actor, two capability provider implementations that satisfy the contracts, and link them together. Note that the actor public key here (starting with `MCUCZ`) assumes you're using the wasmCloud published version of this actor.
+```bash
```
Comment (autodidaddict) [inline ngs/dogs-and-cats/README.md:11]: Do we want to add a warning here that if people are starting these actors from local builds their public keys will be different?

File: ngs/dogs-and-cats/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    async fn handle_request(
+        &self,
+        ctx: &Context,
+        _req: &ServerHttpRequest,
+    ) -> Result<HttpResponse, RpcError> {
+        let client = HttpClientSender::new();
+        // Randomly request a cat or dog image
+        let api_response = match random_in_range(0, 1).await {
+            Ok(0) => client.request(ctx, &HttpRequest::get(CAT_URL)).await?,
+            Ok(1) | _ => client.request(ctx, &HttpRequest::get(DOG_URL)).await?,
```
Comment (autodidaddict) [inline ngs/dogs-and-cats/src/lib.rs:27]: "when all else fails, _dog_"

**Review by autodidaddict — APPROVED** [review]

File: ngs/dogs-and-cats/README.md (hunk 2 above)
Comment (brooksmtownsend) [reply ngs/dogs-and-cats/README.md:11]: Yeah that's what I was going for with the last sentence there, but I will clarify that users will have to change it

Comment (autodidaddict) [reply ngs/dogs-and-cats/README.md:11]: Huh I didn't even notice that there.

**Pull request merged**